OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -420,7 +421,7 @@ public void OnNotifyResourceClaimLost(Actor self, ResourceClaim claim, Actor cla
 
 		PipType GetPipAt(int i)
 		{
-			var n = i * Info.Capacity / Info.PipCount;
+			var n = Info.PipCount >= 0 ? i * Info.Capacity / Info.PipCount : i / -Info.PipCount;
```

Comment:
Likewise here.  Having a harvester that continues to harvest after its pip bar is full is confusing.